### PR TITLE
fix(versioning/loose): Fix suffix comparison

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

- Treat suffix version to be lower than one without suffix

## Context:

- Ref: #renovatebot/docker-swift/pull/37

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/versioning/loose/index.ts [hunk 1]
```diff
@@ -50,12 +50,12 @@ class LooseVersioningApi extends GenericVersioningApi {
...
-    if (!(parsed1.suffix && parsed2.suffix)) {
-      return 1;
+
+    if (parsed1.suffix && parsed2.suffix) {
+      return parsed1.suffix.localeCompare(parsed2.suffix);
     }
-    // equals
-    return parsed1.suffix.localeCompare(parsed2.suffix);
+
+    return this._compareOther(parsed1, parsed2);
```
Comment (zharinov) [inline lib/versioning/loose/index.ts:58]: 0

**Review by rarkins — DISMISSED** [review]

**Review by viceice — COMMENTED** [review]
The strange things at swift is that i use extract version there, so only versions like `5.3` and `5.3.2` should be passed to versioning. 🤔

**Comment by viceice** [comment]
```
[DEBUG] Looking up versions
[DEBUG] Found 44 total versions
[DEBUG] Filter unstable versions
[DEBUG] Found 13 versions within our range
[DEBUG] Candidates: 5.3, 5.3.1, 5.3.2, 5.3.3, 5.4, 5.4.0, 5.4.1, 5.4.2, 5.4.3, 5.5, 5.5.0, 5.5.1, 5.5.2
[DEBUG] Latest stable version is 5.5.2
[DEBUG] Most recent version is 5.5.2
[DEBUG] Build list: 5.3, 5.3.1, 5.3.2, 5.3.3, 5.4, 5.4.0, 5.4.1, 5.4.2, 5.4.3, 5.5, 5.5.0, 5.5.1, 5.5.2
```

**Comment by viceice** [comment]
And we use a start version of `5.3`, so it looks like we should treat longer versions greater that shorter again.

**Review by viceice — CHANGES_REQUESTED** [review]
See comments

File: lib/versioning/loose/index.spec.ts [hunk 2]
```diff
@@ -44,6 +44,10 @@ describe('versioning/loose/index', () => {
     ${'2.4.beta'} | ${'2.4.alpha'} | ${true}
```
Comment (zharinov) [inline lib/versioning/loose/index.spec.ts:44]: > And we use a start version of `5.3`, so it looks like we should treat longer versions greater that shorter again.

But it seems like we already do (see the first test in this table)

@viceice

Comment (viceice) [reply lib/versioning/loose/index.spec.ts:44]: 🤔 here is the code which somehow fails

https://github.com/renovatebot/internal-tools/blob/716a319991bfbe27c5169dd9faed38af0d7634d0/src/commands/docker/builder.ts#L48

Comment (viceice) [reply lib/versioning/loose/index.spec.ts:44]: I'll try to debug that tomorrow from there. Maybe I can the find the issue easier.

**Review by viceice — COMMENTED** [review]
This PR seems to fix the issue, but please add this to tests:

```ts


  test.each`
    a          | b          | expected
    ${'2.4'}   | ${'2.4'}   | ${true}
    ${'2.4.0'} | ${'2.4.0'} | ${true}
    ${'2.4.0'} | ${'2.4'}   | ${false}
    ${'2.4.1'} | ${'2.4'}   | ${false}
    ${'2.4.2'} | ${'2.4.1'} | ${false}
  `('equals("$a", "$b") === $expected', ({ a, b, expected }) => {
    expect(loose.equals(a, b)).toBe(expected);
  });
```

**Review by viceice — APPROVED** [review]

**Pull request merged**